polars 0.54.4 fails to compile (IRFunctionExpr StringExpr / IRStringFunction missing)

### Checks

- [x] I have checked that this issue has not already been reported.
- [x] I have confirmed this bug exists on the [latest version](https://crates.io/crates/polars) of Polars.

### Reproducible example

1. Create project

```bash
cargo new polars-repro
cd polars-repro
```

2. `Cargo.toml`

```toml
[package]
name = "polars-repro"
version = "0.1.0"
edition = "2024"

[dependencies]
polars = { version = "0.54.4", features = ["lazy", "temporal", "timezones"] }
```

3. `src/main.rs`

```rust
fn main() {}
```

4. Run

```bash
cargo check
```

### Log output

```shell
error[E0599]: no variant, associated function, or constant named `StringExpr` found for enum `polars_plan::plans::IRFunctionExpr` in the current scope
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/polars-stream-0.54.4/src/physical_plan/lower_expr.rs:1533:37
     |
1533 |                     IRFunctionExpr::StringExpr(IRStringFunction::Strptime(ref dtype, ref options)),
     |                                     ^^^^^^^^^^ variant, associated function, or constant not found in `polars_plan::plans::IRFunctionExpr`
     |
help: there is a variant with a similar name
     |
1533 -                     IRFunctionExpr::StringExpr(IRStringFunction::Strptime(ref dtype, ref options)),
1533 +                     IRFunctionExpr::StructExpr(IRStringFunction::Strptime(ref dtype, ref options)),
     |

error[E0433]: cannot find type `IRStringFunction` in this scope
    --> /home/user/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/polars-stream-0.54.4/src/physical_plan/lower_expr.rs:1533:48
     |
1533 |                     IRFunctionExpr::StringExpr(IRStringFunction::Strptime(ref dtype, ref options)),
     |                                                ^^^^^^^^^^^^^^^^ use of undeclared type `IRStringFunction`
     |
help: an enum with a similar name exists
     |
1533 -                     IRFunctionExpr::StringExpr(IRStringFunction::Strptime(ref dtype, ref options)),
1533 +                     IRFunctionExpr::StringExpr(IRStructFunction::Strptime(ref dtype, ref options)),
     |

Some errors have detailed explanations: E0433, E0599.
For more information about an error, try `rustc --explain E0433`.
error: could not compile `polars-stream` (lib) due to 2 previous errors
```

### Issue description

polars 0.54.4 fails to compile during build of polars-stream, with internal type/variant mismatches between polars-plan and polars-stream.

The error occurs even in a minimal project and does not depend on user code. It appears to be caused by inconsistent or broken internal API usage in the published crate set for 0.54.4.

This appears related to issues #25492 and #27944, which describe IR / streaming engine inconsistencies in the 0.54

The difference:
- it triggers on a minimal project
- it happens with non-streaming feature sets
- it is reproducible in isolation

### Expected behavior

Project compiles successfully with `polars 0.54.4` and selected features.

### Installed versions

<details>

- cargo 1.96.1 (356927216 2026-06-26)
- rustc 1.96.1 (31fca3adb 2026-06-26
- polars = { version = "0.54.4", features = ["lazy", "temporal", "timezones"] }
</details>

--- a/crates/polars-lazy/Cargo.toml
+++ b/crates/polars-lazy/Cargo.toml
@@ -82,6 +82,7 @@ temporal = [
   "dtype-duration",
   "polars-plan/temporal",
   "polars-expr/temporal",
+  "polars-stream?/temporal",
 ]
 # debugging purposes
 fmt = ["polars-core/fmt", "polars-plan/fmt"]

--- a/crates/polars-stream/Cargo.toml
+++ b/crates/polars-stream/Cargo.toml
@@ -151,7 +151,7 @@ top_k = ["polars-expr/top_k", "polars-plan/top_k"]
 cum_agg = ["polars-expr/cum_agg", "polars-plan/cum_agg", "polars-ops/cum_agg"]
 log = ["polars-expr/log", "polars-plan/log", "polars-ops/log"]
 is_first_distinct = ["polars-core/is_first_distinct", "polars-expr/is_first_distinct", "polars-plan/is_first_distinct"]
-temporal = ["polars-plan/temporal", "polars-expr/temporal"]
+temporal = ["polars-core/temporal", "polars-plan/temporal", "polars-expr/temporal"]
 
 # We need to specify default features here to match workspace defaults.
 # Otherwise we get warnings with cargo check/clippy.

--- a/crates/polars-stream/Cargo.toml
+++ b/crates/polars-stream/Cargo.toml
@@ -151,6 +151,7 @@ top_k = ["polars-expr/top_k", "polars-plan/top_k"]
 cum_agg = ["polars-expr/cum_agg", "polars-plan/cum_agg", "polars-ops/cum_agg"]
 log = ["polars-expr/log", "polars-plan/log", "polars-ops/log"]
 is_first_distinct = ["polars-core/is_first_distinct", "polars-expr/is_first_distinct", "polars-plan/is_first_distinct"]
+temporal = ["polars-plan/temporal", "polars-expr/temporal"]
 
 # We need to specify default features here to match workspace defaults.
 # Otherwise we get warnings with cargo check/clippy.

--- a/crates/polars-stream/src/physical_plan/lower_expr.rs
+++ b/crates/polars-stream/src/physical_plan/lower_expr.rs
@@ -1555,11 +1555,7 @@ fn lower_exprs_with_ctx(
                 transformed_exprs.push(row_idx_col_aexpr);
             },
 
-            #[cfg(any(
-                feature = "dtype-date",
-                feature = "dtype-datetime",
-                feature = "dtype-time"
-            ))]
+            #[cfg(all(feature = "strings", feature = "temporal"))]
             AExpr::Function {
                 input: ref inner_exprs,
                 function:

